### ISSUE TITLE
(JSX) period after anchor tag is moved to newline before break

### ISSUE THREAD

**Issue by SimenB** [prettier/prettier#5047]
**Prettier 1.14.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEkC2AHTAXkwAoBKIgPlIB0pNMAePSuhhxgQ0wAsAnOADNCNEDxgw8aJAHoZcAB6d8AGzgA6XKMwYAnmsLBg2CCoh8kmUQEsoPOH2sxRAGkyDoMAOpxrAc3FLADYABhDMAF8I1np2BhVbAGs2dkYZTkp1FI4AIz5MGRiOGRY6MgBuOhAXEAg8GGt0ZFBOPj4IAHcABVaENGQQTgA3CGsAE2qQPM4wRLgYAGU8Gds-ZBg+AFc4Gts0Bxguvk4-HE5kQU4VfZqAKzQFACFj2fmF5TgAGVs4C6ubkDLPj7PgDHKcHL6aCTPCOWBecYwHjIAAcIRqsIg+y8xzwA1hcBBQ1+NQEAEdNtYBEcTmc-tcdiB9jhrOstoy0Ks1ABFTYQeD0gEwCEIsZI5AAJhqG041gSUD8AGEIDg6SgoNASSBNvsACoQ-pIS4MqJAA)
```sh
--parser babylon
```

**Input:**
```jsx
const comp = () => (
  <p>
    <a href="https://example.com" style={{ color: "inherit", fontWeight: 600 }}>
      link
    </a>.
    <br />
  </p>
);

```

**Output:**
```jsx
const comp = () => (
  <p>
    <a href="https://example.com" style={{ color: "inherit", fontWeight: 600 }}>
      link
    </a>
    .<br />
  </p>
);

```

**Expected behavior:**
For the output to match the input. (`.` after `</a>`, not before `<br />`)

**Comment by duailibe** [comment]
Looks like it was caused by #5006

**Comment by SimenB** [comment]
That was only merged 20 hours ago, though?

**Comment by duailibe** [comment]
You're right 🤦‍♂️

**Comment by SimenB** [comment]
That said, it changed between 1.13.7 and 1.14.0 🙂

**Comment by SimenB** [comment]
@duailibe I bisected, it's from #4717, cc @karl

**Comment by duailibe** [comment]
Thanks @SimenB, I was trying to remember PRs from the top of my head and forgot about that one!

**Comment by karl** [comment]
Good catch! Looks like I missed a test case. I would also have expected the period to stay attached to the a tag.

**Comment by SimenB** [comment]
Maybe special-case a standalone period? It's (always?) a trailing character and not a leading one

**Comment by karl** [comment]
Ah that could work!

I won't have time to work on this for a few days, so if anyone else want to
pick this up please go ahead. Otherwise I'll look at picking it up when I
can.
